projects/plugins/jetpack/modules/masterbar/admin-menu/class-atomic-admin-menu.php

```diff
@@ -144,6 +146,21 @@ public function add_plugins_menu() {
 			$submenus_to_update = array( $slug_to_update => 'https://wordpress.com/plugins/' . $this->domain );
 			$this->update_submenus( 'plugins.php', $submenus_to_update );
 		}
+
+		/**
+		 * Whether to force the calypso plugin page.
+		 * This filter is specific to WPCOM.
+		 *
+		 * @use add_filter( 'wpcom_force_calpyso_plugin_screens', '__return_true' );
+		 * @since 10.6
+		 */
+		if ( apply_filters( 'wpcom_force_calpyso_plugin_screens', false ) ) {
+			$submenus_to_update = array(
+				'plugins.php' => 'https://wordpress.com/plugins/manage/' . $this->domain,
```

Comment:
What about the `plugin-install.php` submenu? If this is assuming that the `wpcom_marketplace_enabled` always returns true (so the `plugin-install.php` submenu is updated above), then I think the code will be more legible if we simply remove the `if ( apply_filters( 'wpcom_marketplace_enabled', false ) )` check.

Also, we could optimize the logic a little bit and avoid 2 calls to `update_submenus`:

```php
if ( ! isset( $submenu['plugins.php'] ) ) {
  return;
}

$plugins_submenu = $submenu['plugins.php'];

// Move "Add New" plugin submenu to the top position.
foreach ( $plugins_submenu as $submenu_key => $submenu_keys ) {
  if ( $submenu_keys[2] === 'plugin-install.php' ) {
    $submenu['plugins.php'] = array( $submenu_key => $plugins_submenu[ $submenu_key ] ) + $plugins_submenu;
  }
}

$submenus_to_update = array( 'plugin-install.php' => 'https://wordpress.com/plugins/' . $this->domain );

if ( apply_filters( 'wpcom_force_calpyso_plugin_screens', false ) ) {
  $submenus_to_update['plugins.php'] = 'https://wordpress.com/plugins/manage/' . $this->domain;
}

$this->update_submenus( 'plugins.php', $submenus_to_update );
```

Reply:
Thanks for the code snippet @mmtr. I've made necessary changes in PR.